src/run.ts

```diff
@@ -52,18 +63,20 @@ export async function run() {
       )
       core.debug(`Writing kubeconfig to ${kubeconfigPath}`)
       const cmd = [
-         'aks',
+         resourceType.toLowerCase()=='microsoft.containerservice/fleets' ? 'fleet':'aks',
          'get-credentials',
          '--resource-group',
          resourceGroupName,
          '--name',
-         clusterName,
+         resourceName,
          '-f',
          kubeconfigPath
       ]
       if (subscription) cmd.push('--subscription', subscription)
-      if (admin) cmd.push('--admin')
-      if (publicFqdn) cmd.push('--public-fqdn')
+      if (resourceType!='fleet') {
+         if (admin) cmd.push('--admin')
```

Comment:
ideally should probably validate above against user setting (admin || publicFqdn) && fleet